route_following_plugin/src/route_following_plugin.cpp

```diff
@@ -411,21 +424,42 @@ void setManeuverLaneletIds(cav_msgs::Maneuver& mvr, lanelet::Id start_id, lanele
             }
             planned_time += getManeuverDuration(latest_maneuver_plan_[i], epsilon_).toSec();
 
-            resp.new_plan.maneuvers.push_back(latest_maneuver_plan_[i]);
+            new_maneuvers.push_back(latest_maneuver_plan_[i]);
             ++i;
         }
 
-        if (resp.new_plan.maneuvers.size() == 0)
+        if (new_maneuvers.size() == 0)
         {
             ROS_WARN_STREAM("Cannot plan maneuver because no route is found");
             return false;
         }
-        //update plan
 
         //Update time progress for maneuvers
-        updateTimeProgress(resp.new_plan.maneuvers, ros::Time::now());
+        if (!req.prior_plan.maneuvers.empty())
+        {
+            updateTimeProgress(new_maneuvers, req.prior_plan.planning_completion_time);
+            ROS_DEBUG_STREAM("Detected a prior plan! Using back maneuver's end time:"<< std::to_string(req.prior_plan.maneuvers.back().lane_following_maneuver.end_time.toSec()));    
+            ROS_DEBUG_STREAM("Where plan_completion_time was:"<< std::to_string(req.prior_plan.planning_completion_time.toSec()));            
+        }
+        else
+        {
+            updateTimeProgress(new_maneuvers, ros::Time::now());
+            ROS_DEBUG_STREAM("Detected NO prior plan! Using ros::Time::now():"<< std::to_string(ros::Time::now().toSec()));   
+        }
+
         //update starting speed of first maneuver
-        updateStartingSpeed(resp.new_plan.maneuvers.front(), current_speed_);
+        updateStartingSpeed(new_maneuvers.front(), req.veh_logitudinal_velocity);
```

Comment:
This should also be in the conditionals above. If there was a prior_plan we should be using end_speed